src/routes/blogs.rs

```diff
@@ -17,19 +17,20 @@ use plume_models::{
     db_conn::DbConn,
     instance::Instance,
     posts::Post,
-    users::User
 };
-use routes::{Page, errors::ErrorPage};
+use routes::{Page, PlumeRocket, errors::ErrorPage};
 use template_utils::Ructe;
-use Searcher;
 
 #[get("/~/<name>?<page>", rank = 2)]
-pub fn details(intl: I18n, name: String, conn: DbConn, user: Option<User>, page: Option<Page>) -> Result<Ructe, ErrorPage> {
+pub fn details(name: String, page: Option<Page>, rockets: PlumeRocket) -> Result<Ructe, ErrorPage> {
     let page = page.unwrap_or_default();
+    let conn = rockets.conn;
     let blog = Blog::find_by_fqn(&*conn, &name)?;
     let posts = Post::blog_page(&*conn, &blog, page.limits())?;
     let articles_count = Post::count_for_blog(&*conn, &blog)?;
     let authors = &blog.list_authors(&*conn)?;
+    let user = rockets.user;
+    let intl = rockets.intl;
```

Comment:
Maybe Rust supports destructuring (I don't remember to be honest, so you could try to do:

```rust
let { user, intl } = rockets;
```

Reply:
Your syntax is not right, but there is a way https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=572b846e985d4780607766e40563e6ba